homeassistant/components/melcloud/sensor.py

```diff
@@ -19,130 +21,148 @@
 from .const import DOMAIN
 
 
-class SensorMetadata(NamedTuple):
-    """Metadata for an individual sensor."""
+@dataclass
+class MelcloudSensorEntityDescription(SensorEntityDescription):
+    """Describes Melcloud sensor entity."""
 
-    measurement_name: str
-    icon: str
-    unit: str
-    device_class: str
-    value_fn: Callable[[Any], float]
-    enabled: Callable[[Any], bool]
+    _value_fn: Callable[[Any], float] | None = None
+    _enabled: Callable[[Any], bool] | None = None
```

Comment:
As these are callables, they could have a default lambda that raises

Reply:
True, but my argument still stands. Raising inside `__post_init__` will fail first: https://github.com/home-assistant/core/pull/53568#discussion_r677723800

This is my suggestion. If you still like me to change it, please say so and I'll do it. In the end it's a waste of both our time for something that shouldn't really happen anyway.

Reply:
I'll DM you :) Let discuss this :)